department-of-veterans-affairs/caseflow
# Include user id when sending data to Google Analytics

Connects #15298

### Description
Add code to send the `User.id` (a consecutive auto-generated number referencing a database record) for the logged-in user as a Google Analytics custom metric called `dimension1`.

See https://github.com/department-of-veterans-affairs/caseflow/issues/15298#issuecomment-698661754

`app/views/layouts/application.html.erb:3` 
renders `app/views/layouts/_head.html.erb:16`, 
which renders `app/views/layouts/_head_google_analytics.html.erb:12`

### Testing Plan
1. Open a caseflow page
2. 'View source' and search for 'dimension1'
3. Verify that code does not interfere with current Caseflow functionality

## Confirmation in prod
1. Deploy to prod; ensuring nothing is broken
2. Check Google Analytics for `db_user_id` custom metric and create a report: https://moz.com/blog/custom-dimensions-google-analytics
